dora/src/bytecode/generator.rs

```diff
@@ -811,6 +811,22 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
                     self.emit_invoke_direct(return_type, return_reg, fct_def_id, num_args);
                 }
             }
+            CallType::ModuleMethod(_, _, _) => {
+                let is_super_call = expr
+                    .object()
+                    .map(|object| object.is_super())
+                    .unwrap_or(false);
+
+                if is_super_call {
+                    self.emit_invoke_direct(return_type, return_reg, fct_def_id, num_args);
+                } else if fct.is_virtual() {
+                    self.emit_invoke_virtual(return_type, return_reg, fct_def_id, num_args);
+                } else if arg_bytecode_types.is_empty() {
```

Comment:
This probably needs to check whether the first argument is a pointer.

Reply:
Thinking about this: Right now there is no singleton object, so there is never self and therefore this should probably always emit self.

Reply:
Yep ... it feels like a lot of these changes are living at an uneasy compromise between what would be sufficient/correct right now vs. what would be the desired behaviors in the future ...